src/SocialServer/Network/PacketHandler.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Melia.Shared.Network;
+using Melia.Shared.Tos.Const;
+using Melia.Social.Database;
+using Yggdrasil.Logging;
+using Yggdrasil.Security.Hashing;
+using static System.Net.Mime.MediaTypeNames;
+
+namespace Melia.Social.Network
+{
+	public class PacketHandler : PacketHandler<ISocialConnection>
+	{
+		/// <summary>
+		/// Sent when entering the game.
+		/// </summary>
+		/// <param name="conn"></param>
+		/// <param name="packet"></param>
+		[PacketHandler(Op.CS_LOGIN)]
+		public void CS_LOGIN(ISocialConnection conn, Packet packet)
+		{
+			var accountName = packet.GetString(56);
+			var password = packet.GetBinAsHex(16);
+			var b1 = packet.GetByte();
+			var accountId = packet.GetLong();
+			var sessionKey = packet.GetString(256);
+
+			// Check account
+			if (!SocialServer.Instance.Database.AccountExists(accountName))
+			{
+				conn.Close(100);
+				return;
+			}
+
+			// Check password
+			var account = Account.LoadFromDb(accountName);
+			if (!BCrypt.CheckPassword(password, account.Password))
+			{
+				conn.Close(100);
+				return;
+			}
+
+			// Logged in
+			conn.Account = account;
+			conn.LoggedIn = true;
+
+			Log.Info("User '{0}' logged in.", conn.Account.Name);
+
+			Send.SC_NORMAL_Unknown_00(conn);
+			Send.SC_LOGIN_OK(conn);
+			Send.SC_NORMAL_Unknown_02(conn);
+		}
+
+		/// <summary>
+		/// Chat
+		/// </summary>
+		/// <param name="conn"></param>
+		/// <param name="packet"></param>
+		[PacketHandler(Op.CS_CHAT)]
+		public void CS_CHAT(ISocialConnection conn, Packet packet)
+		{
+			var size = packet.GetShort();
+			var chat = packet.GetString();
+			var account = conn.Account;
+
+			Log.Debug("CS_CHAT: {0}", chat);
+			// TODO: Save chat and send it after saving it?
+			// TODO: Decide to use ChatCommands?
+			var splitChat = chat.Split(' ');
```

Comment:
If you split strings, make sure to remove empty entries, or your code might break just by someone putting multiple spaces in their message.